File: app/src/main/java/com/jpmcosta/test/realmtestproject/RealmTestProject.java
package com.jpmcosta.test.realmtestproject;

import android.app.Application;
import android.content.Intent;

import androidx.annotation.NonNull;

import com.jpmcosta.test.realmtestproject.realm.App;
import com.jpmcosta.test.realmtestproject.realm.Item;
import com.jpmcosta.test.realmtestproject.realm.Session;
import com.jpmcosta.test.realmtestproject.service.AppUpdateIntentService;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class RealmTestProject extends Application {

    public RealmConfiguration realmConfiguration;


    @Override
    public void onCreate() {
        super.onCreate();

        Realm.init(this);

        realmConfiguration = new RealmConfiguration.Builder()
                .initialData(new Realm.Transaction() {

                    @Override
                    public void execute(@NonNull Realm realm) {
                        realm.copyToRealm(App.create(0L));
                        realm.copyToRealm(Session.create(0L));
                        for (int i = 0; i < 10; i++) {
                            realm.copyToRealm(Item.create((long) i));
                        }
                    }
                })
                .deleteRealmIfMigrationNeeded()
                .build();

        startService(new Intent(this, AppUpdateIntentService.class));
    }
}

File: app/src/main/java/com/jpmcosta/test/realmtestproject/RealmTestProject.kt
package com.jpmcosta.test.realmtestproject

import android.app.Application
import android.os.SystemClock
import com.jpmcosta.test.realmtestproject.realm.RealmInitialData
import io.realm.Realm
import io.realm.RealmConfiguration

class RealmTestProject : Application() {

    lateinit var realmConfiguration: RealmConfiguration


    override fun onCreate() {
        super.onCreate()

        Realm.init(this)

        realmConfiguration = RealmConfiguration.Builder()
            .schemaVersion(SystemClock.elapsedRealtimeNanos()) // Ensure Realm file is recreated.
            .deleteRealmIfMigrationNeeded()
            .initialData(RealmInitialData())
            .build()
    }
}